web/src/Components/StatsPage.js

```diff
@@ -67,10 +86,11 @@ class StatsProvider extends Component {
   renderMain () {
     if (this.state.loading) return (<Loading />)
 
-    const { week, stats } = this.state
+    const { week, filter } = this.state
+    const stats = this.filteredStats(filter, this.state.stats)
 
     return (
-      <div className="container" style={{height: '100%', minHeight: '100%'}}>
+      <div className="container" key={filter} style={{height: '100%', minHeight: '100%'}}>
```

Comment:
is the key required here? Usually key is only so react can keep track of what changes in a list

Reply:
On further reading/re-reading, it seems the issue came from components that maintained the stats in their own state. In that case, the props were only used when creating the component, and it ignored any further changes. Using the props directly in the render function works as expected, without having a key - the Leaderboards work this way.

With that, I think we can leave out the key trick, and have the child components use the props directly - StatsTable can be stateless now, which seems nice.